src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -546,6 +591,127 @@ case object LOG3 extends LogOp(0xa3)
 case object LOG4 extends LogOp(0xa4)
 
 
+sealed abstract class CallOp(code: Int, delta: Int, alpha: Int) extends OpCode(code, delta, 1, G_zero) {
+
+  protected def exec(state: ProgramState): ProgramState = {
+    val (Seq(gas, to, callValue, inAddr, inSize, outAddr, outSize), stack1) = getParams(state)
+
+    val (inputData, mem1) = state.memory.load(inAddr, inSize)
+    val endowment = if (this == DELEGATECALL) BigInt(0) else callValue.toBigInt
+
+    val startGas = {
+      val gExtra = gasExtra(state, endowment, Address(to))
+      val gCap = gasCap(state, gas, gExtra)
+      if (endowment == 0) gCap else gCap + G_callstipend
+    }
+
+    lazy val result = {
+      val toAddr = Address(to)
+
+      val (account, world1) = this match {
+        case CALL =>
+          val w = state.world.transfer(state.ownAddress, toAddr, endowment)
+          val acc = w.getGuaranteedAccount(toAddr)
+          (acc, w)
+
+        case _ =>
+          val acc = state.world.getAccount(toAddr).getOrElse(Account.Empty)
+          (acc, state.world)
+      }
+
+      val (owner, caller) = this match {
+        case CALL => (toAddr, state.ownAddress)
+        case CALLCODE => (state.ownAddress, state.ownAddress)
+        case DELEGATECALL => (state.ownAddress, state.env.callerAddr)
+      }
+
+      val env = state.env.copy(
+        ownerAddr = owner,
+        callerAddr = caller,
+        inputData = inputData,
+        value = callValue,
+        program = Program(world1.getCode(account.codeHash)),
+        callDepth = state.env.callDepth + 1)
+
+      val context = state.context.copy(
+        env = env,
+        startGas = startGas,
+        world = world1)
+
+      VM.run(context)
+    }
+
+    val validCall =
+      state.env.callDepth < OpCode.MaxCallDepth &&
```

Comment:
Yes, we could, although with MaxCallDepth == 1024 that's highly unlikely. Such test could be done after EC-86